src/Service/Sqs/SqsConsumer.php

```diff
@@ -53,9 +62,21 @@ public function handleSqs(SqsEvent $event, Context $context): void
                 $headers[$name] = $attribute['stringValue'];
             }
 
-            $envelope = $this->serializer->decode(['body' => $record->getBody(), 'headers' => $headers]);
+            try {
+                $envelope = $this->serializer->decode(['body' => $record->getBody(), 'headers' => $headers]);
 
-            $this->busDriver->putEnvelopeOnBus($this->bus, $envelope->with(new AmazonSqsReceivedStamp($record->getMessageId())), $this->transportName);
+                $this->busDriver->putEnvelopeOnBus($this->bus, $envelope->with(new AmazonSqsReceivedStamp($record->getMessageId())), $this->transportName);
+            } catch (\Throwable $exception) {
+                if ($this->partialBatchFailure === false) {
+                    throw $exception;
+                }
+
+                if ($this->logger !== null) {
```

Comment:
We could use `NullLogger` by default to avoid this `if` statement

Reply:
Yes of course :+1: